File: nextvisa-client/src/types/reSchedule.ts
export const ScheduleStatus = {
    PENDING: 'PENDING',
    PROCESSING: 'PROCESSING',
    COMPLETED: 'COMPLETED',
    FAILED: 'FAILED',
    NOT_FOUND: 'NOT_FOUND',
    LOGIN_PENDING: 'LOGIN_PENDING',
    SCHEDULED: 'SCHEDULED'
} as const;

export type ScheduleStatus = typeof ScheduleStatus[keyof typeof ScheduleStatus];

export interface ReSchedule {
    id: number;
    applicant: number;
    start_datetime?: string;
    end_datetime?: string;
    status: ScheduleStatus;
    error?: string;
    created_at: string;
    updated_at: string;
}

export interface ReScheduleCreate {
    applicant: number;
    start_datetime?: string;
    end_datetime?: string;
    status?: ScheduleStatus;
    error?: string;
}

export interface ReScheduleUpdate {
    applicant?: number;
    start_datetime?: string;
    end_datetime?: string;
    status?: ScheduleStatus;
    error?: string;
}

File: nextvisa-api/models/re_schedule.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum

class ScheduleStatus(str, Enum):
    """Enum for re-schedule status"""
    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"
    NOT_FOUND = "NOT_FOUND"
    LOGIN_PENDING = "LOGIN_PENDING"
    SCHEDULED = "SCHEDULED"

class ReScheduleBase(BaseModel):
    """Base schema for ReSchedule with common fields"""
    applicant: int = Field(..., gt=0, description="Applicant ID (foreign key)")
    start_datetime: Optional[datetime] = None
    end_datetime: Optional[datetime] = None
    status: ScheduleStatus = ScheduleStatus.PENDING
    error: Optional[str] = None

class ReScheduleCreate(BaseModel):
    """Schema for creating a new re-schedule record"""
    applicant: int = Field(..., gt=0, description="Applicant ID (foreign key)")
    start_datetime: Optional[datetime] = None
    end_datetime: Optional[datetime] = None
    status: ScheduleStatus = ScheduleStatus.PENDING
    error: Optional[str] = None

class ReScheduleUpdate(BaseModel):
    """Schema for updating an existing re-schedule - all fields optional"""
    applicant: Optional[int] = Field(None, gt=0, description="Applicant ID (foreign key)")
    start_datetime: Optional[datetime] = None
    end_datetime: Optional[datetime] = None
    status: Optional[ScheduleStatus] = None
    error: Optional[str] = None

class ReScheduleResponse(BaseModel):
    """Re-schedule response model"""
    id: int
    applicant: int
    start_datetime: Optional[str] = None
    end_datetime: Optional[str] = None
    status: ScheduleStatus
    error: Optional[str] = None
    created_at: str
    updated_at: str
    
    class Config:
        from_attributes = True
